app/assets/src/components/views/discovery/mapping/DiscoveryMap.jsx

```diff
@@ -88,33 +114,49 @@ class DiscoveryMap extends React.Component {
   };
 
   renderMarker = locationInfo => {
-    const { currentTab, previewedLocationId } = this.props;
+    const { currentTab, mapLevel, previewedLocationId } = this.props;
     const { viewport } = this.state;
+
     const id = locationInfo.id;
     const name = locationInfo.name;
     const lat = parseFloat(locationInfo.lat);
     const lng = parseFloat(locationInfo.lng);
+    const geoLevel = locationInfo.geo_level;
+
     const idsField = currentTab === "samples" ? "sample_ids" : "project_ids";
     if (!locationInfo[idsField]) return;
+
     const pointCount = locationInfo[idsField].length;
-    const minSize = 10;
+    // Sizing parameters determined via eyeballing.
+    const minSize = 14;
     // Scale based on the zoom and point count (zoomed-in = higher zoom)
-    // Log1.5 of the count looked nice visually for not getting too large with many points.
+    // Log1.3 of the count looked nice visually for not getting too large with many points.
     const markerSize = Math.max(
-      Math.log(pointCount) / Math.log(1.4) * (get("zoom", viewport) || 3),
+      Math.log(pointCount) / Math.log(1.3) * (get("zoom", viewport) || 3),
       minSize
     );
 
+    let ShapeMarker;
```

Comment:
Can you define a `<ShapeMarker rectangular />` component please? 
This gets much more confusing here.
The shape marker should all the numbers/sizing hidden inside and maybe configurable through props.

Reply:
refactored 👍